aws_topology/stackstate_checks/aws_topology/aws_topology.py

```diff
@@ -82,28 +90,30 @@ def check(self, instance_info):
             )
             return
 
-        try:
-            if not instance_info.state:
-                # Create empty state
-                instance_info.state = State(
-                    {
-                        'last_full_topology': datetime.utcnow().replace(tzinfo=pytz.utc) - timedelta(
-                            seconds=init_config.full_run_interval + 60
-                        )
-                    }
-                )
-            self.delete_ids = []
-            self.components_seen = set()
-            if self.must_run_full(instance_info, init_config.full_run_interval):
-                instance_info.state.last_full_topology = datetime.utcnow().replace(tzinfo=pytz.utc)
+        self.delete_ids = []
+        self.components_seen = set()
+        if self.must_run_full(init_config.full_run_interval):
+            try:
+                self.log.info('Starting FULL topology scan')
+                self.last_full_topology = datetime.utcnow().replace(tzinfo=pytz.utc)
                 self.get_topology(instance_info, aws_client)
+                self.log.info('Finished FULL topology scan (no exceptions)')
+                self.service_check(self.SERVICE_CHECK_EXECUTE_NAME, AgentCheck.OK, tags=instance_info.tags)
+            except Exception as e:
+                msg = "AWS topology collection failed: {}".format(e)
+                self.log.error(msg)
+                self.service_check(
+                    self.SERVICE_CHECK_EXECUTE_NAME, AgentCheck.WARNING, message=msg, tags=instance_info.tags
+                )
+
+        try:
             self.get_topology_update(instance_info, aws_client)
```

Comment:
Do we always run the update straight after the full scan? We can just wait for the next check run right